src/core/AudioEngine.cpp

```diff
@@ -481,24 +479,41 @@ void AudioEngine::handleMetronome()
 	}
 
 	tick_t ticks = song->getPlayPos(currentPlayMode).getTicks();
-	tick_t ticksPerBar = TimePos::ticksPerBar();
-	int numerator = song->getTimeSigModel().getNumerator();
-
-	if (ticks == lastMetroTicks)
+	if (ticks == m_prevMetronomeStates.second)
 	{
 		return;
 	}
 
-	if (ticks % (ticksPerBar / 1) == 0)
+	/* Metronome sound generation - explanation:
+	 * The modulo operation below figures out the exact point in time when a sound should be played. As this operation
+	 * relies on ".getTicks()" we have to add a certain tick tolerance, since it is not guaranted that between each call
+	 * the tick gets a +1-increment and thus the exact point in time can be missed.*/	
+
+	// increase this parameter if with complex projects or poor HW resources the metronome rings out of rythm
+	int nToleranceTicks = 5;
+	// LMMS allows abrupt jumps in song positions (e.g. song-loops, when marker was put of that loop), this iscached below
+	bool noDoubleTick = m_prevMetronomeStates.first > nToleranceTicks;
+	// identify which sound should be played below
+	bool highTickEvent = ticks % int(TimePos::ticksPerBar() / 1) <= nToleranceTicks;
```

Comment:
Why `/ 1`? This is just `TimePos::ticksPerBar()`.

Reply:
That's the lingo in master: https://github.com/LMMS/lmms/blob/e3044575bce939cb00989a4cce6163c2111d3f8c/src/core/AudioEngine.cpp#L492

Reply:
It's literally fluff. Even at `-O0` it's removed. Let's not copy this.

Reply:
Yep compiler kills it, as @zonkmachine  mentioned I just took it over from current master. I considered as a reminder for the dev how the timeslots are choosen. I don't have any hard feelings for it.